ivy/functional/backends/torch/linear_algebra.py

```diff
@@ -425,27 +425,22 @@ def svdvals(x: torch.Tensor, /, *, out: Optional[torch.Tensor] = None) -> torch.
 
 # ToDo: re-add int32 support once
 # (https://github.com/pytorch/pytorch/issues/84530) is fixed
-@with_unsupported_dtypes({"2.0.1 and below": ("int32",)}, backend_version)
+@with_unsupported_dtypes(
+    {"2.0.1 and below": ("int32", "bool", "float16")}, backend_version
+)
 def tensordot(
     x1: torch.Tensor,
     x2: torch.Tensor,
     /,
     *,
     axes: Union[int, Tuple[List[int], List[int]]] = 2,
+    batched_modes: Optional[Union[int, Tuple[List[int], List[int]]]] = None,
     out: Optional[torch.Tensor] = None,
 ) -> torch.Tensor:
-    # find the type to promote to
-    dtype = ivy.as_native_dtype(ivy.promote_types(x1.dtype, x2.dtype))
-    # type conversion to one that torch.tensordot can work with
-    x1, x2 = x1.type(torch.float32), x2.type(torch.float32)
+    return torch.tensordot(x1, x2, dims=axes)
 
-    # handle tensordot for axes==0
-    # otherwise call with axes
-    if axes == 0:
-        ret = (x1.reshape(x1.size() + (1,) * x2.dim()) * x2).type(dtype)
-    else:
-        ret = torch.tensordot(x1, x2, dims=axes).type(dtype)
-    return ret
+
+tensordot.partial_mixed_handler = _batched_modes_is_none
```

Comment:
Let's directly add the lambda function here instead of keeping it in another file. This enhances readability and will be helpful when the user is debugging their code.

Reply:
understood

Reply:
Yeah, I understand your concern as well but having it in the backend implementation will be very useful for debugging.

Reply:
understood